pkg/netpol/scan/scan.go

```diff
@@ -164,69 +258,241 @@ func searchDeploymentManifests(repoDir string, walkFn WalkFunction) []string {
 	return yamls
 }
 
+// given YAML of kind "List" , parse and convert to slice of K8sObject
+func parseList(objDataBuf []byte) []K8sObject {
+	for kind := range singleResourceK8sKinds {
+		if isKind, resList := parseListOfKind(objDataBuf, kind); isKind {
+			return resList
+		}
+	}
+	return nil
+}
+
+// given YAML of kind "NamespaceList" , parse and convert to slice of K8sObject
 func parseNamespaceList(objDataBuf []byte) []K8sObject {
-	r := bytes.NewReader(objDataBuf)
-	res := make([]K8sObject, 0)
-	if r == nil {
-		return res
+	if isKind, resList := parseListOfKind(objDataBuf, Namespace); isKind {
+		return resList
 	}
-	nsList := v1.NamespaceList{}
-	if err := yaml.NewYAMLOrJSONDecoder(r, yamlParseBufferSize).Decode(&nsList); err == nil {
-		for i := range nsList.Items {
-			nsList.Items[i].Kind = Namespace
-			res = append(res, K8sObject{Namespace: &nsList.Items[i], Kind: Namespace})
+	return nil
+}
+
+// given YAML of kind "PodList" , parse and convert to slice of K8sObject
+func parsePodList(objDataBuf []byte) []K8sObject {
+	if isKind, resList := parseListOfKind(objDataBuf, Pod); isKind {
+		return resList
+	}
+	return nil
+}
```

Comment:
Why are pods and namespaces so special, they get a `parse*List` of their own? Is it because of the tests we have?
I believe all the kinds we care about can have their *List.
In general, `kubectl get` always returns an object of type `List`; It's API GET calls that return the `*List` kinds.
See [here](https://kubernetes.io/docs/reference/using-api/api-concepts/#collections)

Reply:
Yes, it's because of the tests that we have, one test requires support in PodsList and NamespaceList.
I wrote in a TODO comment that we should add support for other `*List` kinds.